core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -161,28 +162,24 @@ public void run() {
         }
         try {
           InetSocketAddress destination = InetSocketAddress.createUnresolved(host, port);
-          ProxyParameters proxy = proxyDetector.proxyFor(destination);
+          final ProxyParameters proxy;
+          try {
+            proxy = proxyDetector.proxyFor(destination);
+          } catch (Exception e) {
+            rescheduleWithCause(savedListener, e);
+            return;
+          }
           if (proxy != null) {
-            EquivalentAddressGroup server = new EquivalentAddressGroup(destination);
+            EquivalentAddressGroup server =
+                new EquivalentAddressGroup(new ProxiedInetSocketAddress(destination, proxy));
```

Comment:
ProxiedInetSocketAddress is weird, and it can't be internal long-term. Any NR should be able to support proxies.

The discussion about it should probably include @zhangkun83. (IIRC, the problem here is how should flattening by pick-first impact the attributes of the `EquivalentAddressGroup`.)

Reply:
ProxiedInetSocketAddress was my suggestion. I don't see a problem making it public if we need to. Can you elaborate your concerns?

Reply:
@zhangkun83 and I talked at length about this offline. It seems we're going to be mucking with how NameResolvers and LBs communicate again. That will need some other requirements to be gathered. But in the short term we want some sort of `PairSocketAddress` (I'm saying that name because it is obviously bad; we need a different name) that would contain `(SocketAddress, Attr)`. It basically shoe-horns something similar to the deleted `ResolvedServerInfo` without a full redesign of the APIs.

Reply:
Does this imply `EquivalentAddressGroup::getAddresses()` would return a list of `PairSocketAddress`? Sounds like we will just commit to making this class public now, and reusing it later for NameResolvers+LBs.

Reply:
It would continue to return a list of `SocketAddress`. No, we're not ready to make this public. There needs to be a longer-term design and that needs some investigation into requirements.

Reply:
ack, changed it to `PairSocketAddress`